### PR TITLE
Bg 72158 safeprime generation 2

### PR DESCRIPTION
**Pull request by rlutzbitgo**
TICKET: BG-72158

# Please be aware of the following when making your pull request:

## Description

We need to generate safe primes for ECDSA TSS signing. This is supported natively by the crypto lib in Node versions >=15. Node 14 is EOL in a few weeks, so this PR gets a head start on that by removing support for Node 14 and only supporting 16 (while we take the time to check support for Node versions [16, 18]). It also adds support for very fast generation of safe primes in the use of ntilde creation for ECDSA TSS. 

I plan on attempting to add support for 18 in a fast follow-up.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [X] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts
Comment (sachushaji) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts:20]: We can also reject the err incase an error appears

File: modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts
Comment (sachushaji) [inline modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts:170]: `createKeccakHash` should be implementing the `Hash` interface, Causing some issues?

Comment (rlutzbitgo) [reply modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts:170]: Yeah, there's insufficient overlap, it's missing the property "copy" or something? Not sure if this is just a types issue, but tests still pass so I think this should work. I can revisit it when I try and add 18 support, or if anyone else wants to look into this that would be great too.

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts
Comment (rlutzbitgo) [reply modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts:20]: Yeah that's fair. I'll add that before I try and wrangle approvals

**Review by LeoKotschenreuther — CHANGES_REQUESTED** [review]
Sounds like you are doing two things at once. Please split your PR up into two commits (or even two PRs if you like). The first commit should be only the Node upgrade, the second should be your feature change.

**Comment by rlutzbitgo** [comment]
> Sounds like you are doing two things at once. Please split your PR up into two commits (or even two PRs if you like). The first commit should be only the Node upgrade, the second should be your feature change.

I kind of wanted to keep these linked in case this goes disastrously and needs reverted. The safe prime generation via native crypto lib will only work on Node versions >=15.

**Review by mohammadalfaiyazbitgo — APPROVED** [review]
LGTM, will need to test how this integrates with the react app once released.

**Review by jicli — APPROVED** [review]

File: modules/account-lib/test/unit/mpc/tss/ecdsa/rangeproof.ts
Comment (LeoKotschenreuther) [inline modules/account-lib/test/unit/mpc/tss/ecdsa/rangeproof.ts:15]: Why add this test if you skip it?

**Review by LeoKotschenreuther — APPROVED** [review]
NodeJS changes LGTM

File: modules/account-lib/test/unit/mpc/tss/ecdsa/rangeproof.ts
Comment (rlutzbitgo) [reply modules/account-lib/test/unit/mpc/tss/ecdsa/rangeproof.ts:15]: It works locally, but was having issues with it in CI. Was going to address in a follow-up

**Review by wesgraham — APPROVED** [review]

**Review by olibeyene — CHANGES_REQUESTED** [review]
do we need to update bgms for this to work? i.e. new version of hsm 

Can you verify sending tx works?

**Review by olibeyene — APPROVED** [review]

**Comment by rlutzbitgo** [comment]
I'm going to hold off on merging this till I ensure it plays nicely with WP

**Pull request closed without merging**
